api/routes/profile.js

```diff
@@ -163,29 +163,33 @@ router.post('/', authRequired, (req, res) => {
   const profile = req.body;
   if (profile) {
     const id = profile.id || 0;
-    Profiles.findBy({ id })
-      .first()
-      .then((pf) => {
-        if (pf == undefined) {
-          //profile not found so lets insert it
-          Profiles.create(profile)
-            .then((profile) =>
-              res
-                .status(200)
-                .json({ message: 'profile created', profile: profile[0] })
-            )
-            .catch((err) => {
-              console.error(err);
-              res.status(500).json({ message: err.message });
-            });
-        } else {
-          res.status(400).json({ message: 'profile already exists' });
-        }
-      })
-      .catch((err) => {
-        console.error(err);
-        res.status(500).json({ message: err.message });
-      });
+    try {
+      Profiles.findById(id)
+        .then((pf) => {
+          if (pf == undefined) {
+            //profile not found so lets insert it
+            Profiles.create(profile)
+              .then((profile) =>
+                res
+                  .status(200)
+                  .json({ message: 'profile created', profile: profile[0] })
+              )
+              .catch((err) => {
+                console.error(err);
+                res.status(500).json({ message: err.message });
+              });
+          } else {
+            res.status(400).json({ message: 'profile already exists' });
+          }
+        })
+        .catch((err) => {
+          console.error(err);
+          res.status(500).json({ message: err.message });
+        });
```

Comment:
all these inner catches should bubble up to the top of the try. We can probably remove them to shorten the code a lot.

Reply:
I haven't tested this. Can you confirm.

Reply:
I haven't tested this code specifically but that's behavior I've used before. It's part of the spec. 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises#Error_propagation

Reply:
Looking over this again it looks like you aren't using async here. I don't think that try catch will actually catch anything because the inner promise isn't being resolved. Only the inner catches would get anything.
I'll take some time later and write up an example of what that would look like.

Reply:
I wrote how this would look and submitted a PR. I tested before and after and get the same error messages so the propagation is working. 

https://github.com/Lambda-School-Labs/betterreads-be-lx.a/pull/4